serialization.rst

```diff
@@ -222,6 +225,62 @@ possible mapping values are:
   if it exists, will be sent as an associative array to the ``bsonUnserialize``
   function to initialise the object's properties.
 
+Scalar Types
+------------
+
+For types that the driver has a wrapper class for, it is possible to define a
+map from each supported data type to a class of your own, providing it
+implements the ``MongoDB\BSON\TypeWrapper`` interface.
+
+The supported data types are Binary, Decimal128, Javascript, MaxKey, MinKey,
+ObjectID, Regex, Timestamp, and UTCDateTime.
+
+The ``MongoDB\BSON\TypeWrapper`` interface defines two functions:
+``createFromBSONType()``, a factory method which takes a ``MongoDB\BSON\Type``
+argument, and ``toBSONType()``, which returns a ``MongoDB\BSON\Type`` value.
+
+As an example, a wrapped UTCDateTime class, could look like::
+
+    class UTCDateTimeWrapper implements \MongoDB\BSON\TypeWrapper
+    {
+        private $intern;
+
+        public function __construct( \MongoDB\BSON\UTCDateTime $type )
+        {
+            $this->intern = $type->toDateTime();
+        }
+
+        static function createFromBSONType(\MongoDB\BSON\Type $type)
+        {
+            if (! $type instanceof \MongoDB\BSON\UTCDateTime) {
+                throw UnexpectedValueException;
+            }
+
+            return new UTCDateTimeWrapper( $type );
+        }
+
+        function toBSONType() : \MongoDB\BSON\Type
+        {
+            return new \MongoDB\BSON\UTCDateTime( $this->intern );
+        }
+    }
+
+
+If the defined class wraps (composes) an original ``MongoDB\BSON\*`` class,
+then it SHOULD also implement the accompanying
+``\MongoDB\BSON\<classname>Interface`` interface.
+
+The type specific interfaces include all of the methods from the original
+class, with the exact same arguments and return types.
+
+For example, a user-defined ``UTCDateTimeWrapper`` class needs to implement the
+``MongoDB\BSON\UTCDateTimeInterface`` and ``MongoDB\BSON\TypeWrapper``
+interfaces.
+
+It is not necessary to return an object from the ``createFromBSONType``
+"factory" method. You may also return a simple scalar value.
```

Comment:
I realize you *can* return a scalar, since you're just going to inject it into the PHP representation of the BSON document, but why would you do this? I imagine it would leave you with no way to round-trip data. I could perhaps see this used as a way to downgrade Decimal128 values to regular floating points or integers.